dali/pipeline/operators/reader/loader/sequence_loader.cc

```diff
@@ -32,37 +32,9 @@ std::string parent_dir(std::string path) {
 
 }  // namespace
 
-void SequenceLoader::PrepareEmpty(TensorSequence *sequence) {
-  sequence->tensors.resize(sequence_length_);
-  for (auto &t : sequence->tensors) {
-    PrepareEmptyTensor(&t);
-  }
-}
+namespace filesystem {
 
-void SequenceLoader::ReadSample(TensorSequence *sequence) {
-  // TODO(klecki) this is written as a prototype for video handling
-  const auto &stream = streams_[current_stream_];
-  // TODO(klecki) we probably should buffer the "stream", or recently used
-  // frames
-  for (int i = 0; i < sequence_length_; i++) {
-    LoadFrame(stream, current_frame_ + i, &sequence->tensors[i]);
-  }
-  current_frame_++;
-  // wrap-around
-  if (current_frame_ == stream_sizes_[current_stream_]) {
-    current_stream_++;
-    current_frame_ = 0;
-  }
-  if (current_stream_ == streams_.size()) {
-    current_stream_ = 0;
-  }
-}
-
-Index SequenceLoader::Size() {
-  return total_size_;
-}
-
-std::vector<SequenceLoader::Stream> SequenceLoader::ParseStreams(string file_root) {
+std::vector<Stream> GatherExtractedStreams(string file_root) {
   glob_t glob_buff;
   std::string glob_pattern = file_root + "/*/*";
```

Comment:
Maybe we should filter out unknown file extensions?
Also you could think about file_list alike functionality - that user could just provide list of frames and paths to them.

Reply:
I do not put requirements on the particular extensions, whatever the ImageFactory parses, is ok. 

For the second one, I will add a task: https://jirasw.nvidia.com/browse/DALI-445

Reply:
Done